src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,534 @@
+        require(msg.sender == gameCreator || _isValidMetaTx(gameCreator), "EDITOR_ACCESS_DENIED");
+        _setGameEditor(gameCreator, editor, isEditor);
+    }
+
+    /// @notice Transfers creatorship of `original` from `sender` to `to`.
+    /// @param sender The address of current registered creator.
+    /// @param original The address of the original creator whose creation are saved in the ids themselves.
+    /// @param to The address which will be given creatorship for all tokens originally minted by `original`.
+    function transferCreatorship(
+        address sender,
+        address original,
+        address to
+    ) external override notToZero(to) {
+        require(
+            msg.sender == sender || _isValidMetaTx(sender) || _superOperators[msg.sender],
+            "TRANSFER_ACCESS_DENIED"
+        );
+        require(sender != address(0), "NOT_FROM_ZEROADDRESS");
+        address current = _creatorship[original];
+        if (current == address(0)) {
+            current = original;
+        }
+        require(current != to, "CURRENT_=_TO");
+        require(current == sender, "CURRENT_!=_SENDER");
+        if (to == original) {
+            _creatorship[original] = address(0);
+        } else {
+            _creatorship[original] = to;
+        }
+        emit CreatorshipTransfer(original, current, to);
+    }
+
+    /// @notice Create a new GAME token.
+    /// @param from The address of the one creating the game (may be different from msg.sender if metaTx).
+    /// @param to The address who will be assigned ownership of this game.
+    /// @param creation The struct containing ids & ammounts of assets to add to this game,
+    /// along with the uri to set.
+    /// @param editor The address to allow to edit (can also be set later).
+    /// @param subId A random id created on the backend.
+    /// @return id The id of the new GAME token (erc721).
+    function createGame(
+        address from,
+        address to,
+        Update calldata creation,
+        address editor,
+        uint64 subId
+    ) external override onlyMinter() notToZero(to) notToThis(to) returns (uint256 id) {
+        (uint256 gameId, uint256 baseId) = _mintGame(from, to, subId, 0, true);
+
+        if (editor != address(0)) {
+            _setGameEditor(to, editor, true);
+        }
+        if (creation.assetIdsToAdd.length != 0) {
+            _addAssets(from, baseId, creation.assetIdsToAdd, creation.assetAmountsToAdd);
+        }
+
+        _metaData[baseId] = creation.uri;
+        _numNFTPerAddress[to]++;
+        emit GameTokenUpdated(0, gameId, creation, to);
+        return gameId;
+    }
+
+    /// @notice Get game editor status.
+    /// @param gameOwner The address of the owner of the GAME.
+    /// @param editor The address of the editor to set.
+    /// @return isEditor Editor status of editor for given tokenId.
+    function isGameEditor(address gameOwner, address editor) external view override returns (bool isEditor) {
```

Comment:
Now that game editor act on the whole set of token, we could remove them and use ApprovalForAll operator instead.

Reply:
@wighawag We could. I tend to think of the ApprovalForAll operator as having slightly different permissions...
For example, currently, a `gameEditor` can modify games, but _**cannot**_ transfer them.
I'm fine to implement the change if we're certain that this is what we want.

Reply:
This is true and since such game editor could be another human, this is probably better.
In that case, we should restrict the game editor to send the new Game token to a different address, else it would have the same power as ApprovalForAll

Reply:
As it is now, the editor can only set the `to` address when removing assets from a game, and this is where the assets are sent to. The new game token which is created is not transferred at any time unless `transfer` is called explicitly.

Reply:
Changed so that all withdrawn assets return to owner address. marking as resolved, please unresolved if this doesn't meet our needs.